src/fixate/drivers/dmm/fluke_8846a.py

```diff
@@ -268,10 +273,54 @@ def voltage_dc(self, _range=None, auto_impedance=False):
             command = "; :SENS:VOLT:DC:IMP:AUTO OFF"
         self._set_measurement_mode("voltage_dc", _range, suffix=command)
 
-    def current_ac(self, _range=None):
+    def current_ac(self, _range, port):
+        """
+        Set the measurement mode on the DMM to AC current.
+
+        If the range and port selection are not compatible, i.e. someone has requested to measure
+        1 A on the low range port with a maximum capability of 400 mA, an exception is raised.
+
+        If the range requested can be measured by the low port, but the high port is selected, an
+        exception is raised.
+        """
+
+        # Check the requested range is not more than the port capability:
+        if _range > self.current_ports[port]:
+            raise ValueError(
+                "The selected port and range combination is not available for this instrument. Consider using a different multimeter"
+            )
+
+        # Raise an error if the high port is selected when the low port should be used:
+        if _range < self.current_ports["LOW"] and port == "HIGH":
+            raise ValueError(
+                "High range port selected when the low range port should be used! Consider using a different multimeter."
```

Comment:
Shouldn't the consideration here be to use the low current port?

Reply:
Is that not what it is doing already? If a small range is selected, and the high port requested, the driver will throw an exception requesting that the low range port should be used instead.

This is because the DMM will often select the low range port automatically based on the given range, so I thought it helpful to force the port selection to reduce unexpected behavior.